File: lab/python/textos/apenas_digitos_com_regex.py
import re


def isdigit(value):
    if isinstance(value, (int, float, complex)):
        return "SIM"

    value = str(value).strip()
    pattern = r"^-?\d+(?:[\.,]\d+)?$"  # aceita ponto ou vírgula decimal
    return "SIM" if re.match(pattern, value) else "NÃO"


# Testes
print(isdigit("12345"))          # SIM
print(isdigit(12345))            # SIM
print(isdigit(complex(12345)))   # SIM
print(isdigit("-12345"))         # SIM
print(isdigit("12345.5"))        # SIM
print(isdigit("12345,5"))        # SIM
print(isdigit("  12345 "))       # SIM
print(isdigit("123a45"))         # NÃO
print(isdigit("-123a45"))        # NÃO
print(isdigit("-12 345"))        # NÃO
print(isdigit("abcde"))          # NÃO
print(isdigit("-abcde"))         # NÃO

File: lab/javascript/textos/apenas_digitos_com_regex.js
function isDigit(value) {
  // se já for tipo numérico (number), é SIM
  if (typeof value === "number") return "SIM";

  // tenta converter valores do tipo objeto Number ou complexos (simulados)
  if (value instanceof Number) return "SIM";

  // força para string e remove espaços
  const str = String(value).trim();

  // Regex: aceita inteiros e decimais com ponto ou vírgula
  const pattern = /^-?\d+(?:[.,]\d+)?$/;

  return pattern.test(str) ? "SIM" : "NÃO";
}

// Testes
console.log(isDigit("12345"));      // SIM
console.log(isDigit(12345));        // SIM
console.log(isDigit("-12345"));     // SIM
console.log(isDigit("12345.5"));    // SIM
console.log(isDigit("12345,5"));    // SIM
console.log(isDigit("  12345 "));   // SIM
console.log(isDigit("123a45"));     // NÃO
console.log(isDigit("-123a45"));    // NÃO
console.log(isDigit("-12 345"));    // NÃO
console.log(isDigit("abcde"));      // NÃO
console.log(isDigit("-abcde"));     // NÃO
